src/Parser/Internal/AttributesDocBlockParser.php

```diff
@@ -19,53 +18,19 @@ public function __construct(CommandInfo $commandInfo, \ReflectionMethod $reflect
     {
         $this->commandInfo = $commandInfo;
         $this->reflection = $reflection;
+        // @todo Unused. Lets just remove from this class?
         $this->fqcnCache = $fqcnCache ?: new FullyQualifiedClassCache();
     }
 
+    /**
+     * Call the handle method of each attribute, which alters the CommandInfo object.
+     */
     public function parse()
     {
         $attributes = $this->reflection->getAttributes();
         foreach ($attributes as $attribute) {
-            if ($attribute->getName() === self::COMMAND_ATTRIBUTE_CLASS_NAME) {
-                foreach ($attribute->getArguments() as $argName => $argValue) {
-                    switch ($argName) {
-                        case 'name':
-                            $this->commandInfo->setName($argValue);
-                            break;
-                        case 'description':
-                            $this->commandInfo->setDescription($argValue);
-                            break;
-                        case 'help':
-                            $this->commandInfo->setHelp($argValue);
-                            break;
-                        case 'aliases':
-                            $this->commandInfo->setAliases($argValue);
-                            break;
-                        case 'usage':
-                            $this->commandInfo->setExampleUsage(key($argValue), array_pop($argValue));
-                            break;
-                        case 'options':
-                            $set = $this->commandInfo->options();
-                            foreach ($argValue as $name => $option) {
-                                $description = trim(preg_replace('#[ \t\n\r]+#', ' ', $option['description']));
-                                $this->commandInfo->addOptionDescription($name, $description);
-                            }
-                            break;
-                        case 'params':
-                            $set = $this->commandInfo->arguments();
-                            foreach ($argValue as $name => $param) {
-                                $description = trim(preg_replace('#[ \t\n\r]+#', ' ', $param['description']));
-                                $this->commandInfo->addArgumentDescription($name, $description);
-                            }
-                            break;
-                        default:
-                            foreach ($argValue as $name => $annotation) {
-                                foreach ($annotation as $value) {
-                                    $this->commandInfo->addAnnotation($name, $value);
-                                }
-                            }
-                    }
-                }
+            if (in_array(AttributeInterface::class, class_implements($attribute->getName()))) {
```

Comment:
I think this should just be a simple `method_exists`. Then you can just get rid of `AttributeInterface`. I'm not sure that overriding static methods works right in PHP; maybe it does, but a) it's unusual, and b) it's not really what you're doing with `call_user_func` below. I think that the static method solution is fine, but let's not conflate it with an OO dispatch.

Reply:
Good idea. Done.